src/agent/custom-http-agent.ts

```diff
@@ -210,4 +222,21 @@ export class CustomHttpAgent {
       makeNonceTransform((): Nonce => base64ToUint8Array(nonce) as Nonce)
     );
   }
+  private attachAddTransformExpiry(expiry: number): void {
+    this.#agent.addTransform('update', makeExpiryTransform(expiry));
+  }
+
+  private getIngressExpiry(hash: string): number {
+    const existingExpiry = this.#cache.get(hash);
+
+    if (existingExpiry && !isTimestampExpired(expiryToMs(existingExpiry))) {
```

Comment:
Can you add a TODO about what to do if the the timestamp is expired? Not sure if we should try with a new expiry or reject the request.